pkg/apis/eventing/v1alpha1/trigger_types.go

```diff
@@ -58,11 +58,10 @@ type TriggerSpec struct {
 	// +optional
 	DeprecatedGeneration int64 `json:"generation,omitempty"`
 
-	Broker     string               `json:"broker,omitempty"`
-	Subscriber *SubscriberSpec      `json:"subscriber,omitempty"`
+	Broker     string          `json:"broker,omitempty"`
+	Subscriber *SubscriberSpec `json:"subscriber,omitempty"`
 
-	Type   string `json:"type,omitempty"`
-	Source string `json:"source,omitempty"`
+	Filters *metav1.LabelSelector `json:"filters,omitempty"`
```

Comment:
Let's put in another layer of indirection:

```yaml
spec:
  filter:
    headers:
      Ce-Eventtype: com.example.someevent
      ...
```

That gives an easy place to add something like `headerExpressions`, as sibling of `headers`

Reply:
Done. Having headerExpressions will be sligthly more involved than I thought. There is a Requirement struct that doesn't expose its fields. I'm trying to do a quick prototype using reflection to set those values. For now, I just pushed the headers with exact matching. It works as what you previously had of type and source.